.babelrc.js

```diff
@@ -69,35 +72,43 @@ var babelConfig = {
   plugins
 }
 
-if (env === 'browser') {
-  babelConfig = Object.assign(babelConfig, {
-    presets: [['@babel/preset-env', browserBabelPresetEnvConfig]]
-  })
-}
+module.exports = function (api) {
 
-if (env === 'legacy') {
-  babelConfig = Object.assign(babelConfig, {
-    presets: [['@babel/preset-env', legacyBabelPresetEnvConfig]]
-  })
-}
+  var env = api.env()
 
-if (env === 'modules') {
-  babelConfig = Object.assign(babelConfig, {
-    presets: [['@babel/preset-env', modulesBabelPresetEnvConfig]]
-  })
-}
+  if (env === 'browser') {
+    babelConfig = Object.assign({}, babelConfig, {
+      presets: [['@babel/preset-env', browserBabelPresetEnvConfig]]
+    })
+  }
 
-if (env === 'node') {
-  babelConfig = Object.assign(babelConfig, {
-    presets: [['@babel/preset-env', nodeBabelPresetEnvConfig]]
-  })
-}
+  if (env === 'legacy') {
+    babelConfig = Object.assign({}, babelConfig, {
+      presets: [['@babel/preset-env', legacyBabelPresetEnvConfig]]
+    })
+  }
 
-if (env === 'test') {
-  babelConfig = Object.assign(babelConfig, {
-    presets: [['@babel/preset-env', testBabelPresetEnvConfig]],
-    plugins: babelConfig.plugins.concat(['rewire'])
-  })
-}
+  if (env === 'modules') {
+    babelConfig = Object.assign({}, babelConfig, {
+      presets: [['@babel/preset-env', modulesBabelPresetEnvConfig]]
+    })
+  }
+
+  if (env === 'node') {
+    babelConfig = Object.assign({}, babelConfig, {
+      presets: [['@babel/preset-env', nodeBabelPresetEnvConfig]]
+    })
+  }
 
-module.exports = babelConfig
+  if (env === 'test') {
+    babelConfig = Object.assign({}, babelConfig, {
+      presets: [['@babel/preset-env', testBabelPresetEnvConfig]],
+      plugins: babelConfig.plugins.concat(['rewire'])
+    })
+  }
+
+  console.debug('BABEL ENV: ' + env)
```

Comment:
do we want these logs in there?

Reply:
This is a helpful information because it prints the current target (`browser`, `node`, `test`, `legacy`) to the screen. In case of errors, it's obvious which target failed.